<launch>
    <param name="max-depth-demand"    value="10.0" />                  <!-- Maximum depth demand (m) -->
    <param name="over-depth"          value="12.0" />                  <!-- Maximum depth (m) - causes abort -->
    <param name="over-pitch"          value="45" />                   <!-- Maximum pitch (deg) - causes abort -->
    <param name="over-roll"           value="90" />                   <!-- Maximum roll (deg) - causes abort -->
    <param name="max-internal-temp"   value="45" />                   <!-- Maximum internal pressure vessel temp (deg) - causes abort -->
    <param name="min-motor-voltage"   value="19000" />                <!-- Minimum Motor voltage (mV) - causes abort -->
    <param name="mission-timeout"     value="20" />                   <!-- Mission timeout (minutes) -->
    <param name="lat_orig"            value="54.4300000" />        <!-- Origin Latitude -->
    <param name="long_orig"           value="-7.68373666667000" />            <!-- Origin Longitude -->
    <param name="look_ahead_distance" value="10" />                   <!-- look ahead distance (in m) for track following algorithm  effectivly inverse of gain-->    
    

    <param name="StartX"              value="-30.0" />                  <!-- Start X location (m) -->
    <param name="StartY"              value="0.0" />                  <!-- Start Y location (m) -->
    <param name="FinishX"             value="-40.0" />                  <!-- Finish X location (m) -->
    <param name="FinishY"             value="0.0" />                  <!-- Finish Y location (m) -->
    
    <!--Sub will Return to this Location-->
    <param name="HomeX"               value="-30.0" />                  <!-- Home X location (m) -->
    <param name="HomeY"               value="0.0" />                  <!-- Home Y location (m) -->
    
    
    <param name="MinAltitude"         value="1.5" />                  <!-- Start X location (m) -->
    <param name="Speed"               value="0.8" />                  <!-- Start X location (m) -->
    <param name="n"               value="3" />                  <!-- Nos of spaces -->
    
    <!--Notes-->
    <!--Testwood lat_orig=50.93642234975332 long_orig=-1.507850997377956 -->
    <!--Based on this origin X=12 and Y=15 is a good location for HOME at Testwood -->
    
    <!--Easleigh South lat_orig=50.95646666666666 long_orig=-1.366823 -->
    <!--Coordinates are for south pier, middle south of lake (X,Y) = (-32.3,22.6) -->
    
    <!--Easleigh North lat_orig=50.95702499999999 long_orig=-1.36675104 -->
    <!--Coordinates are for south pier, middle south of lake (X,Y) = (-24.6,30.0) -->
    

    <node name="mission_script" pkg="DelphinROSv2" type="mission_script_probe.py"  output="screen" required="true" /> 
    
    <!-- Add all other nodes under here! -->
    
    <node name="back_seat_driver" pkg="DelphinROSv2" type="BackSeatDriver.py" required="true" />
	
    <!-- <node name="compass_dummy" pkg="DelphinROSv2" type="compass_dummy.py" /> -->
    <node name="arduino" pkg="DelphinROSv2" type="arduino.py" /> 
    <node name="compass_oceanserver" pkg="DelphinROSv2" type="compass_oceanserver.py" /> 
    <node name="temp_sensor" pkg="DelphinROSv2" type="temp_sensor.py" /> 
    <node name="gps" pkg="DelphinROSv2" type="gps_nmea2.py" /> 
    <node name="altimeter" pkg="DelphinROSv2" type="altimeter_micron.py" /> 
    <node name="Cameras" pkg="DelphinROSv2" type="camera.py" output = "screen"/>
    <!-- <node name="SMS_modem" pkg="DelphinROSv2" type="SMS_modem.py" /> -->
    <include file="$(find DelphinROSv2)/launch/sonar_system.launch" />
    
    <node name="dead_reckoner" pkg="DelphinROSv2" type="dead_reckoner4.py"  /> 
    
    <!-- <node name="depth_controller" pkg="DelphinROSv2" type="depth_controller.py" output = "screen"/> -->
    <node name="heading_controller" pkg="DelphinROSv2" type="heading_controller.py"/>
    
    <!-- <node name="heading_controller_MPC_v1" pkg="DelphinROSv2" type="heading_controller_MPC_nonlinear_v1.py"/> -->
    <node name="depthandspeed_controller_MPC_v5" pkg="DelphinROSv2" type="depthandspeed_controller_MPC_nonlinear_v5.py" />
        
    <include file="$(find DelphinROSv2)/launch/thruster_system.launch" />
    <node name="tail_section" pkg="DelphinROSv2" type="tail_section.py"/> 
    
    <node name="logger" pkg="DelphinROSv2" type="logger.py" output="screen"/> 
    <!-- <node name="smach_viewer" pkg="smach_viewer" type="smach_viewer.py" /> -->

    

</launch>
